src/psyclone/psyir/nodes/psy_data_node.py

```diff
@@ -123,18 +125,20 @@ def __init__(self, ast=None, children=None, parent=None, options=None):
         super(PSyDataNode, self).__init__(ast=ast, children=[sched],
                                           parent=parent)
 
-        # Store the name of the PSyData variable that is used for this
-        # PSyDataNode. This allows the variable name to be shown in str
-        # (and also, calling create_name in gen() would result in the name
-        # being changed every time gen() is called).
+        # Get or create a symbol table so we can avoid name clashes
+        # when creating variables
         if parent and hasattr(self.root, 'symbol_table'):
             symtab = self.root.symbol_table
         else:
             # FIXME: This may not be a good solution
             symtab = SymbolTable()
 
+        # Store the name of the PSyData variable that is used for this
+        # PSyDataNode. This allows the variable name to be shown in str
+        # (and also, calling create_name in gen() would result in the name
+        # being changed every time gen() is called).
         self._var_name = symtab.new_symbol_name(
-            self.make_symbol(self.psy_data_var))
+            self.add_psydata_class_prefix("psy_data"))
```

Comment:
Could this line use `self._psy_data_symbol_with_prefix`?

Reply:
Yes, indeed.